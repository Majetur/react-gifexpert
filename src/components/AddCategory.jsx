import { useState } from "react"

export const AddCategory = ({onNewCategory}) => {

  //const [inputValue, setInputValue] = useState('One Punch')
  const [inputValue, setInputValue] = useState('')

/*   const onInputChange = (event) =>{
    console.log(event.target.value);
    setInputValue(event.target.value)
  }
 */
  const onInputChange = ({target}) =>{
    setInputValue(target.value)
  }

  const onSubmit = (event) =>{
    event.preventDefault()
    if(inputValue.trim().length<=1) return
    onNewCategory(inputValue.trim())
    setInputValue('')
    //console.log(inputValue)
  }

  return (
    <form onSubmit={onSubmit}>
        <input 
            type="text"
            placeholder="Buscar gif"
            value={inputValue}
            /* onChange={(event) => onInputChange (event)} */
            onChange={onInputChange}
        />
    </form>
  )
}
