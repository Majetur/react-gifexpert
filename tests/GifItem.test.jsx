import { render,screen, waitFor  } from "@testing-library/react"
import { GifItem } from "../src/components/GifItem"

describe('Pruebas con componente GitItem', () => { 
    
    const title = "prueba"
    const url = "urlPrueba"
    test('debe coincidir con el snapshot del componente', () => { 
        const {container} = render (<GifItem title={title} url={url}/>)
        expect(container).toMatchSnapshot()
     })

    test('debe mostrar la imagen con el URL y el alt indicado', () => {    
        
        render (<GifItem title={title} url={url}/>)
        screen.debug()
        expect(screen.getByRole('img').src).toBe(url)
        


     })
     
 })