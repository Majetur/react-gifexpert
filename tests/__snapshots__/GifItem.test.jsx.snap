// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con componente GitItem debe coincidir con el snapshot del componente 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="prueba"
      src="urlPrueba"
    />
    <p>
      prueba
    </p>
  </div>
</div>
`;
